### PR TITLE
feat(#248): Restricting roles from rooms

### LINKED ISSUE #248: Restrict Access of Users from Rooms
### User story
As an admin, I can restrict the access of certain rooms from certain user groups.
### Narrative
When making a room, an admin should be able to limit who has access to it. For example, if there is something that is only allowed to be used by people who work at or go to Concordia, then no one with a role outside of Concordia may book the room. 
### Out of scope
NaN
### Acceptance criteria

| # | Given | When | Then |
|---|-------|------|------|
|1|As an admin|when i restrict a type of user for a room|that user can no longer book a room|
|2|As a user|when my role is restricted from booking a room|the user is unable to make a booking request|
|3|As a user|when my role is not restricted from booking a room|the user is able to make a booking request|
|4|As an admin|when updating the restrictions of a room|bookings already made won't be cancelled|

### PR DESCRIPTION
**Pull request by EvanDime**
This MR is for adding the feature of limiting access of certain roles from specific rooms. When merged, users who are not able to book certain rooms won't have them appear in their search results or if they manage to get a link to the page, submitting the form will block them form making the booking request. 

### Test plan (required)
Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.

Feature test showing a scope and making sure it hides rooms user is not allowed to book. 
Feature test showing that the api allowing to change restrictions for rooms works
UI test to see if you are able to change the restrictions for a room. 
### Closing issues
List all issues this PR resolves.
- closes #248

### Pre-Merge checklist 
Complete these items before requesting reviews
- [ ] Static Analysis Rules Validated
- [x] All commits follow naming conventions 

### Sub-tasks Checklist
- [x] UI Implementation
- [x] UI Test
- [ ] Backend Implementation
- [ ] Backend Tests
- [ ] Integration Tests
- [ ] System Tests
- [ ] Documentation

### CODE REVIEW

File: routes/web.php [hunk 1]
```diff
@@ -27,9 +28,9 @@
     Route::resource('users', UserController::class)->only(['store', 'index', 'destroy', 'update']);
 
     Route::resource('roles',\App\Http\Controllers\RoleController::class)->except(['show', 'edit']);
-
     Route::resource('rooms',\App\Http\Controllers\RoomController::class)->only(['store', 'index', 'update', 'destroy']);
-    Route::resource('rooms',\App\Http\Controllers\RoomController::class)->only(['store', 'index', 'update','destroy']);
+    Route::put('room/restrictions/{id}', RestrictionsController::class.'@update')
```
Comment (WindOfRussia) [inline routes/web.php:32]: I believe the approach is now to pass an array of `[controller::class, 'method_name']`

File: app/Http/Controllers/BookingRequestController.php [hunk 2]
```diff
@@ -22,7 +22,7 @@ public function index()
     {
         return inertia('Admin/BookingRequests/Index', [
             'booking_requests' => BookingRequest::with('user', 'room')->get(),
-            'rooms' => Room::all(),
+            'rooms' => Room::hideUserRestrictions(auth()->user())->get(),
```
Comment (WindOfRussia) [inline app/Http/Controllers/BookingRequestController.php:25]: should probably pull user off of $request->user rather than relying on a global method.

**Review by WindOfRussia — CHANGES_REQUESTED** [review]
few small changes but great overall

**Review by KeeganWoods1 — COMMENTED** [review]
Code looks great, feature works well, just the unit test to look into and I'll approve

**Comment by EvanDime** [comment]
> Code looks great, feature works well, just the unit test to look into and I'll approve
@KeeganWoods1 What exactly am i missing as far as unit test is concerned? Last i checked i had 100% coverage testing on all pieces of backend

**Review by KeeganWoods1 — DISMISSED** [review]
Everything works, all tests pass!

**Review by WindOfRussia — CHANGES_REQUESTED** [review]
all good, tests passing but it would be a good idea to split up that massive test into 3 smaller ones

**Comment by WindOfRussia** [comment]
fyi circleCI is down rn so I was testing locally.

File: tests/Feature/RoomRestrictionsTest.php [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+class RoomRestrictionsTest extends TestCase
+{
+    use RefreshDatabase;
+
+    /**
+     * Test if the route can create, update and remove restrictions from the system
+     *
+     * @return void
+     */
+    public function test_restrictions_controller()
```
[the remark this answers is not in the document]
Comment (EvanDime) [reply tests/Feature/RoomRestrictionsTest.php]: Latest commit has the refactor in place.

**Review by WindOfRussia — APPROVED** [review]

**Review by KeeganWoods1 — APPROVED** [review]

**Pull request merged**